omim2obo/parsers/omim_entry_parser.py

```diff
@@ -406,3 +414,44 @@ def get_self_ref_assocs(phenotype_mim: str, gene_phenotypes: Dict[str, Dict]) ->
         if not _assoc['phenotype_mim_number']:
             _self_ref_assocs.append(_assoc)
     return _self_ref_assocs
+
+
+def _add_to_review_tsv(class_code: int, value: str):
+    """Update REVIEW_CASES with review cases, which will later be written to review.tsv"""
+    REVIEW_CASES.append({
+        "classCode": class_code,
+        "classShortName": REVIEW_CASE_NAME_MAP[class_code],
+        "value": value,
+    })
+
+
+def log_review_cases(
+    p_mim: str, p_lab: str, p_map_key: str, gene_mim: str, gene_phenotypes: Dict[str, Dict],
+    omim_types: Dict[str, str]
+):
+    """Log cases that need to be reviewed"""
+    global REVIEW_SELF_REF_CASE_I
+    p_lab_lower: str = p_lab.lower()
+    basic_review_info = f"(Phenotype: {p_mim} {p_lab}), (Map key: {p_map_key}), (Gene: {gene_mim})"
+
+    # - Digenic: Should technically be none marked 'digenic' if only 1 association, but there are.
+    if 'digenic' in p_lab_lower:
+        _add_to_review_tsv(1, basic_review_info)
+    # = Somatic mutations
+    if 'somatic' in p_lab_lower:
+        _add_to_review_tsv(3, basic_review_info)
+    # - Self-referential cases
+    self_ref_assocs: List[Dict] = get_self_ref_assocs(p_mim, gene_phenotypes)
+    if self_ref_assocs:
+        REVIEW_SELF_REF_CASE_I += 1
+        _add_to_review_tsv(2, f"{REVIEW_SELF_REF_CASE_I}: {basic_review_info}")
+    for self_ref_assoc in self_ref_assocs:
+        _add_to_review_tsv(2, f"{REVIEW_SELF_REF_CASE_I}: (Phenotype: {self_ref_assoc['phenotype_label']}), (Map key: "
+            f"{self_ref_assoc['phenotype_mapping_info_key']}), (Gene: {p_mim})", )
+    # - Unexpected non-phenotype MIM types
+    p_mim_type: str = omim_types[p_mim]  # Allowable: PHENOTYPE, HERITABLE_PHENOTYPIC_MARKER (#, %)
+    mim_type_err = f"(Phenotype MIM type {p_mim_type}), {basic_review_info}"
+    if p_mim_type == 'GENE':  # *
```

Comment:
nit: Since the `#` is a symbol in OMIM that has meaning, you might want to add some text after the `#` used as a code comment to separate out the code comment symbol from the OMIM symbol as you've done on [Line 452](https://github.com/monarch-initiative/omim/pull/178/files#diff-8e252738feb17a9d3e230d6603290660b2cb0503be65d6ad66b5b0675901489dR452). I'm used to reading this now, but still takes a few seconds to realize what is the OMIM symbol in the comment 😄

Reply:
Sensible! I will change it to:
> `# Represented by: *`